esp/src/src-react/components/SubFiles.tsx

```diff
@@ -101,26 +102,35 @@ export const SubFiles: React.FunctionComponent<SubFilesProps> = ({
         message: nlsHPCC.RemoveSubfiles2,
         items: selection.map(item => item.Name),
         onSubmit: React.useCallback(() => {
-            WsDfu.SuperfileAction("remove", file.Name, selection, false).then(() => refresh());
-        }, [file, refresh, selection])
+            WsDfu.SuperfileAction("remove", file.Name, selection, false).then(() => {
+                refreshSubfiles();
+                if (file.subfiles.Item.length - 1 === 0) {
+                    replaceUrl(`/files/${file.Name}`);
```

Comment:
Is this to refresh the host component?  Might be a race ondition on the `page refresh` and the `refreshSubfiles`?